src/main/java/com/lesson7/solution/service/impl/TodoServiceImpl.java

```diff
@@ -0,0 +1,66 @@
+package com.lesson7.solution.service.impl;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.lesson7.solution.entity.TaskHistory;
+import com.lesson7.solution.entity.Todo;
+import com.lesson7.solution.exception.EntityNotFoundException;
+import com.lesson7.solution.mapper.TodoMapper;
+import com.lesson7.solution.model.TodoCreateDto;
+import com.lesson7.solution.model.TodoResponseDto;
+import com.lesson7.solution.model.TodoUpdateDto;
+import com.lesson7.solution.repository.TaskHistoryRepository;
+import com.lesson7.solution.repository.TodoRepository;
+import com.lesson7.solution.service.TodoService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+@RequiredArgsConstructor
+@Service
+public class TodoServiceImpl implements TodoService {
+    private final TodoRepository todoRepository;
+    private final TaskHistoryRepository taskHistoryRepository;
+    private final TodoMapper todoMapper;
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    @Override
+    public TodoResponseDto create(TodoCreateDto todoCreateDto) {
+        return todoMapper.toResponseDto(todoRepository.save(todoMapper.createToModel(todoCreateDto)));
+    }
+
+    @Transactional
+    @Override
+    public TodoResponseDto update(Long id, TodoUpdateDto todoUpdateDto) {
+        Todo todoOld = todoRepository.findById(id).orElse(null);
+        if (todoOld == null) {
+            throw new EntityNotFoundException("Todo with id " + id + " not found.");
+        }
+        Todo todoNew = todoMapper.updateToModel(todoUpdateDto);
```

Comment:
Use `orElseThrow` instead of fetching a nullable value and then checking it.

```suggestion
    public TodoResponseDto update(Long id, TodoUpdateDto todoUpdateDto) {
        Todo todoOld = todoRepository.findById(id)
            .orElseThrow(() -> new EntityNotFoundException("Todo with id " + id + " not found."));
        Todo todoNew = todoMapper.updateToModel(todoUpdateDto);
```